plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -640,19 +640,80 @@ public function get_total_shipping_refunded( $order ) {
 		return 0;
 	}
 
+	/**
+	 * Finds an Order ID based on an order key.
+	 *
+	 * @param string $order_key An order key has generated by.
+	 * @return int The ID of an order, or 0 if the order could not be found
+	 */
 	public function get_order_id_by_order_key( $order_key ) {
-		// TODO: Implement get_order_id_by_order_key() method.
-		return 0;
+		global $wpdb;
+
+		$orders_table = self::get_orders_table_name();
+		$op_table     = self::get_operational_data_table_name();
+
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		return $wpdb->get_var(
+			$wpdb->prepare(
+				"SELECT {$orders_table}.id FROM {$orders_table}
+				INNER JOIN {$op_table} ON {$op_table}.order_id = {$orders_table}.id
+				WHERE {$op_table}.order_key = %s",
+				$order_key
+			)
+		);
+		// phpcs:enable
 	}
 
+	/**
+	 * Return count of orders with a specific status.
+	 *
+	 * @param  string $status Order status. Function wc_get_order_statuses() returns a list of valid statuses.
+	 * @return int
+	 */
 	public function get_order_count( $status ) {
-		// TODO: Implement get_order_count() method.
-		return 0;
+		global $wpdb;
+
+		$orders_table = self::get_orders_table_name();
+
+		return absint( $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(*) FROM {$orders_table} WHERE type = %s AND status = %s", 'shop_order', $status ) ) ); // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+	}
+
+	/**
+	 * Get all orders matching the passed in args.
+	 *
+	 * @deprecated 3.1.0 - Use {@see wc_get_orders} instead.
+	 * @param  array $args List of args passed to wc_get_orders().
+	 * @return array|object
+	 */
+	public function get_orders( $args = array() ) {
+		wc_deprecated_function( __FUNCTION__, '3.1.0', 'Use wc_get_orders instead.' );
```

Comment:
What do you think of:

```suggestion
		wc_deprecated_function( __METHOD__, '3.1.0', 'Use wc_get_orders instead.' );
```

Just because it is a little more specific (it gives us something like `ClassName::method_name` instead of just `method_name`).